src/modules/graph/graph.service.ts

```diff
@@ -0,0 +1,27 @@
+import { ClientSecretCredential } from '@azure/identity';
+import { Client } from '@microsoft/microsoft-graph-client';
+import { TokenCredentialAuthenticationProvider } from '@microsoft/microsoft-graph-client/authProviders/azureTokenCredentials';
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import GraphConfig from '@ukef/config/graph.config';
+
+type RequiredConfigKeys = 'clientId' | 'clientSecret' | 'tenantId' | 'scope';
+
+@Injectable()
+export class GraphService {
+  client: Client;
+  constructor(
+    @Inject(GraphConfig.KEY)
+    private readonly config: Pick<ConfigType<typeof GraphConfig>, RequiredConfigKeys>,
```

Comment:
I don't think we need `private readonly` here as we don't need to keep the `config` as a field. That also means you could destructure the values immediately if you want

Reply:
Great -- And that means I can also remove the `Pick` as it's unneeded